pwnlib/util/misc.py

```diff
@@ -195,10 +197,14 @@ def run_in_new_terminal(command, terminal = None, args = None):
         - If X11 is detected (by the presence of the ``$DISPLAY`` environment
           variable), ``x-terminal-emulator`` is used.
 
+    If `kill_at_exit` is :const:`True`, try to close the command/terminal when the
+    current process exits. This may not work for all terminal types.
```

Comment:
It is a little unfortunate that this doesn't work nicely for all terminal types. One way to do it would be to wrap the command in a shell script which dumps its pid to a temp file before execing the command. I didn't want to think too hard about shell escaping, which is why I've settled for making it just work in tmux here. Happy to try out a more general approach if you think it'd be useful though.

Reply:
I guess that it should work quite ok. For shell escaping, there is an excellent [`sh_string`](https://docs.pwntools.com/en/stable/util/sh_string.html) module in pwntools, so you don't have to think so hard :)